### PR TITLE
Expose settings through HostSettings instead of ValidateNServiceBusSettings that happens too late for certain transports

### PR DESCRIPTION
**Pull request by mauroservienti**

### CODE REVIEW

File: src/NServiceBus.Core/Transports/HostSettings.cs [hunk 1]
```diff
@@ -10,15 +11,23 @@ public class HostSettings
         /// <summary>
         /// Creates a new instance of <see cref="HostSettings"/>.
         /// </summary>
-        public HostSettings(string name, string hostDisplayName, StartupDiagnosticEntries startupDiagnostic, Action<string, Exception> criticalErrorAction, bool setupInfrastructure)
+        public HostSettings(string name, string hostDisplayName, StartupDiagnosticEntries startupDiagnostic, Action<string, Exception> criticalErrorAction,
+            bool setupInfrastructure, SettingsHolder coreSettings)
```
Comment (timbussmann) [inline src/NServiceBus.Core/Transports/HostSettings.cs:15]:
```suggestion
        public HostSettings(string name, string hostDisplayName, StartupDiagnosticEntries startupDiagnostic, Action<string, Exception> criticalErrorAction, bool setupInfrastructure, SettingsHolder coreSettings)
```

Comment (timbussmann) [reply src/NServiceBus.Core/Transports/HostSettings.cs:15]: should `coreSettings` be an optional parameter?

**Review by timbussmann — APPROVED** [review]
see my comments for my recommended changes but feel free to adjust & merge or merge without the changes and we can discuss my points later.

File: src/NServiceBus.Core/Transports/HostSettings.cs (hunk 1 above)
Comment (mauroservienti) [reply src/NServiceBus.Core/Transports/HostSettings.cs]: Makes sense

**Review by tmasternak — APPROVED** [review]

**Pull request merged**